# Optimización de ganancia
El mercado de divisas es muy grande, El comercio diario a menudo supera los $ 1 trillón R El mercado interbancario mundial.
Los instrumentos derivados de FX incluyen: Opciones de moneda
Gestionar posiciones de moneda para operaciones internacionales.
Gestionar riesgo de moneda corporativa
 Carteras de inversión global.
 Especular en los mercados de divisas.
Riesgo de la moneda de cobertura en inversiones de capital internacional, Cobertura / especulación en mercados globales de renta fija.
Una transacción de moneda al contado es un acuerdo para comprar cierta cantidad de
una moneda usando otra moneda
  + Buscar la manera más optima de obtener la ganancía deseada.
  Con las transacciones que sean necesarias, buscamos llegar a la ganancía obtenida. 
  
  Siendo el mismo resultado el que nos arroje, con cuanto dinero queremos comenzar las transacciones y cuanto queremos recuperar, en la misma moneda. 
  
  
  
  
## Objetivo general

   + Buscamos con este es proyecto es obtener 1,000 pesos mexicanos como ganancía después de comprar y vender monedas al tipo de cambio del día de hoy. 
## Objetivos especificos

 ### Definición de proyecto
   + Definir las variables.
   + Definir las restricciones.
   + Definir la función a maximizar.
 ### Programado
   + Aplicaremos los aprendizajes obtenidos en el modulo 1, con respecto a la programación lineal y resolverlo programando con python. 

## Las modenas y su tipo de cambio que tomaremos en cuenta son las siguientes:

![imagen.png](TablaC.png)


Definición de variables


- $x_1$: Pesos Mexicanos a Dolares
- $x_2$: Pesos Mexicanos a Euros
- $x_3$: Pesos Mexicanos a Yen
- $x_4$: Pesos Mexicanos a Libras esterlinas
- $x_5$: Dolares a Pesos Mexicanos
- $x_6$: Dolares a Euro
- $x_7$: Dolares a Yen
- $x_8$: Dolares a Libres esterlinas
- $x_9$: Euro a Peso Mexicano
- $x_10$: Euro a Dólar
- $x_11$: Euro a Yen
- $x_12$: Euro a Libras esterlinas
- $x_13$: Yen a Pesos Mexicanos
- $x_14$: Yen a Euro 
- $x_15$: Yen a Libras Esterlinas
- $x_16$: Libras esterlinas a Pesos Mexicanos
- $x_17$: Libras esterlinas a Dolares
- $x_18$: Libras esterlinas a Euros
- $x_19$: Libras esterlinas a Yen


# Función a maximizar

f1

# Restricciones:
f1>=1000

In [1]:
import sympy as sym #Importamos librerias
import numpy as np
from scipy.optimize import linprog

In [2]:
f1 = np.array([[-1,-1,-1,-1,0.052,0,0,0,0.046,0,0,0,5.62,0,0,0,0.041,0,0,0]])
f2 = np.array([[19.1,0,0,0,-1,-1,-1,-1,0,0.87,0,0,0,107.32,0,0,0,0.79,0,0]])
f3 = np.array([[0,21.77,0,0,0,1.14,0,0,-1,-1,-1,-1,0,0,122.3,0,0,0,0.88,0]])        
f4 = np.array([[0,0,0.18,0,0,0,0.0093,0,0,0,0.0083,0,-1,-1,-1,-1,0,0,0,0.0073]])
f5 = np.array([[0,0,0,24.38,0,0,0,1.28,0,0,0,1.12,0,0,0,136.74,-1,-1,-1,-1]])


In [3]:
A = np.concatenate ((-f1,-f2), axis=0)
A = np.concatenate ((A,-f3), axis=0)
A = np.concatenate ((A,-f4), axis=0)
A = np.concatenate ((A,-f5), axis=0)
A = np.concatenate ((A, f1), axis=0)


In [6]:
b = np.array([[0,0,0,0,0,1000]])

In [7]:
resultado = linprog(-f1[0],A_ub=A,b_ub=b)
resultado

     con: array([], dtype=float64)
     fun: -999.9999999999854
 message: 'Optimization terminated successfully.'
     nit: 7
   slack: array([1.00000000e+03, 0.00000000e+00, 0.00000000e+00, 0.00000000e+00,
       0.00000000e+00, 1.45519152e-11])
  status: 0
 success: True
       x: array([    0.        ,     0.        , 86206.89655172,     0.        ,
           0.        ,     0.        ,     0.        ,     0.        ,
           0.        ,     0.        ,     0.        ,     0.        ,
       15517.24137931,     0.        ,     0.        ,     0.        ,
           0.        ,     0.        ,     0.        ,     0.        ])

# Ruta sugerida

1) Empezamos con 86206.89655172 pesos mexicanos cambiandolos Yen, y después convertimos de euro a yen pesos mexicanos. 